app/controllers/wishlists_for_guests_controller.rb

```diff
@@ -0,0 +1,10 @@
+class WishlistsForGuestsController < ApplicationController
+  def show
+    existing_wishlist = Wishlist.find_by(token: params[:token])
+      if existing_wishlist
```

Comment:
You can write
`@wishlist = Wishlist.find_by(token: params[:token])`
`render file: 'public/404' unless @wishlist`